src/models/data.py

```diff
@@ -184,15 +194,28 @@ def calculate_normalizing_array(self, data_vars: List[str]):
         for var in data_vars:
             for norm_var in normalizing_dict_keys:
                 if var.endswith(norm_var):
-                    mean.append(self.normalizing_dict[norm_var]['mean'])
-                    std.append(self.normalizing_dict[norm_var]['std'])
+                    var_mean = self.normalizing_dict[norm_var]['mean']
+                    var_std = self.normalizing_dict[norm_var]['std']
+                    if yearly_mean:
+                        var_mean = np.mean(var_mean)
```

Comment:
this is definitely across all of the data and not the batches right?

Reply:
how does this differ from the `yearly_mean` defined below on L230:

```python
        yearly_mean = x.mean(dim=['time', 'lat', 'lon'])
        yearly_mean_np = yearly_mean.to_array().values

        x = self._add_extra_dims(x, self.surrounding_pixels, self.monthly_means)


```

Reply:
What happens on line 200 is calculation of the `normalizing array` for the `yearly_mean_array` calculated in line 230

Reply:
gotcha!